x-pack/plugins/security_solution/server/endpoint/routes/actions/audit_log_handler.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TypeOf } from '@kbn/config-schema';
+import { RequestHandler } from 'kibana/server';
+import { AGENT_ACTIONS_INDEX } from '../../../../../fleet/common';
+import { EndpointActionLogRequestSchema } from '../../../../common/endpoint/schema/actions';
+
+import { SecuritySolutionRequestHandlerContext } from '../../../types';
+import { EndpointAppContext } from '../../types';
+
+export const actionsLogRequestHandler = (
+  _endpointContext: EndpointAppContext
+): RequestHandler<
+  TypeOf<typeof EndpointActionLogRequestSchema.params>,
+  unknown,
+  unknown,
+  SecuritySolutionRequestHandlerContext
+> => {
+  return async (context, req, res) => {
+    const esClient = context.core.elasticsearch.client.asCurrentUser;
+    let result;
+    try {
+      result = await esClient.search({
+        index: AGENT_ACTIONS_INDEX,
+        body: {
+          query: {
+            match: {
+              agents: req.params.agent_id,
+            },
+          },
+          sort: [
+            {
+              '@timestamp': {
+                order: 'desc',
+              },
+            },
+          ],
+        },
+      });
+    } catch (error) {
+      return res.customError({
+        statusCode: 500,
+        body: {
+          message: error,
+        },
+      });
+    }
+    if (result?.statusCode !== 200) {
+      return res.customError({
```

Comment:
No need to use `customError()` - kibana takes care of `500` errors if you just throw.

Suggestion:

```typescript
throw new Error(`Error fetching actions log for agent_id ${req.params.agent_id}`);
```

Also, since this is a controlled failure and we're not returning anything about the actual error that happen in elasticsearch, you should log the error to kibana logs. You can get a logger instance from the endpoint context I think.

Reply:
Does Kibana take care of `4xx` errors as well?

Reply:
done 8d46a982516a15bbc22f8d763e56d43a36b91e6c